src/coredump/coredump.c

```diff
@@ -1063,9 +1066,24 @@ static int context_parse_iovw(Context *context, struct iovec_wrapper *iovw) {
                 if (context->pidref.pid != parsed_pid)
                         return log_error_errno(r, "Passed PID " PID_FMT " does not match passed " PID_FMT ": %m", parsed_pid, context->pidref.pid);
         } else {
-                r = pidref_set_pid(&context->pidref, parsed_pid);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to initialize pidref from pid " PID_FMT ": %m", parsed_pid);
+                if (!isempty(context->meta[META_ARGV_PIDFD]))  {
+                        int fd;
+
+                        r = safe_atoi(context->meta[META_ARGV_PIDFD], &fd);
+                        if (fd < 0)
```

Comment:
`if (r < 0)`

Reply:
pidref_set_pidfd_consume already checks for negative